tests/templates/test_layers.py

```diff
@@ -642,3 +644,226 @@ def circuit(weights):
+            return qml.expval(qml.PauliZ(0))
+
+        with pytest.raises(ValueError, match=msg_match):
+            circuit(weights=weights, wires=wires, init_state=np.array([1, 1, 0, 0]))
+
+    def test_integration(self, tol):
+        """Test integration with PennyLane"""
+
+        N = 4
+        wires = range(N)
+        layers = 2
+        weights = np.array(
+            [
+                [[-0.09009989, -0.00090317], [-0.16034551, -0.13278097], [-0.02926428, 0.05175079]],
+                [[-0.07988132, 0.11315495], [-0.16079166, 0.09439518], [-0.04321269, 0.13678911]],
+            ]
+        )
+
+        dev = qml.device("default.qubit", wires=N)
+
+        @qml.qnode(dev)
+        def circuit(weights):
+            ParticleConservingU1(weights, wires, init_state=np.array([1, 1, 0, 0]))
+
+        return [qml.expval(qml.PauliZ(w)) for w in range(N)]
+
+        res = circuit(weights)
+
+        exp = np.array([-0.99993177, -0.9853332, 0.98531251, 0.99995246])
+        assert np.allclose(res, np.array(expected), atol=tol)
+
+    @pytest.mark.parametrize(
+        ("init_state", "exp_state"),
+        [
+            (np.array([0, 0]), np.array([1.0 + 0.0j, 0.0 + 0.0j, 0.0 + 0.0j, 0.0 + 0.0j])),
+            (
+                np.array([1, 1]),
+                np.array(
+                    [
+                        0.00000000e00 + 0.00000000e00j,
+                        1.85571042e-18 - 3.88036599e-20j,
+                        1.22429852e-16 - 1.27988911e-18j,
+                        1.00000000e00 - 1.73472348e-18j,
+                    ]
+                ),
+            ),
+            (
+                np.array([0, 1]),
+                np.array(
+                    [
+                        0.00000000e00 + 0.00000000e00j,
+                        9.99540465e-01 + 0.00000000e00j,
+                        -3.03110196e-02 - 3.16873242e-04j,
+                        1.85571042e-18 + 3.88036599e-20j,
+                    ]
+                ),
+            ),
+            (
+                np.array([1, 0]),
+                np.array(
+                    [
+                        0.00000000e00 + 0.00000000e00j,
+                        -3.03110196e-02 + 3.16873242e-04j,
+                        -9.99540465e-01 + 0.00000000e00j,
+                        1.22429852e-16 + 1.27988911e-18j,
+                    ]
+                ),
+            ),
+        ],
+    )
+    def test_decomposition_u1ex(self, init_state, exp_state, tol):
+        """Test the decomposition of the U_{1, ex}` exchange gate by asserting the prepared
+        state"""
+
+        N = 2
+        wires = range(N)
+        layers = 1
+        weights = np.array([[[0.01045368, -0.03031732]]])
+
+        dev = qml.device("default.qubit", wires=N)
+
+        @qml.qnode(dev)
+        def circuit(weights):
+            ParticleConservingU1(weights, wires, init_state=init_state)
+
+        return qml.expval(qml.PauliZ(0))
```

Comment:
Similarly, need to consider the indentation here.

Reply:
done